eng/mgmt/automation/generate_utils.py

```diff
@@ -334,58 +338,66 @@ def generate_typespec_project(
-            check_call(cmd, sdk_root)
+            check_call(tsp_cmd, sdk_root)
 
             sdk_folder = find_sdk_folder(sdk_root)
             logging.info('SDK folder: ' + sdk_folder)
             if sdk_folder:
+                # parse service and module
+                match = re.match(r'sdk[\\/](.*)[\\/](.*)', sdk_folder)
+                service = match.group(1)
+                module = match.group(2)
+                # check require_sdk_integration
+                cmd = ['git', 'add', '.']
+                check_call(cmd, sdk_root)
+                cmd = ['git', 'status', '--porcelain', os.path.join(sdk_folder, 'pom.xml')]
+                logging.info('Command line: ' + ' '.join(cmd))
+                output = subprocess.check_output(cmd, cwd=sdk_root)
+                output_str = str(output, 'utf-8')
+                git_items = output_str.splitlines()
+                if len(git_items) > 0:
+                    git_pom_item = git_items[0]
+                    # new pom.xml implies new SDK
+                    require_sdk_integration = git_pom_item.startswith('A ')
+                if not require_sdk_integration and remove_before_regen and group_id:
+                    # clear existing generated source code, and regenerate
+                    drop_changes(sdk_root)
+                    remove_generated_source_code(sdk_folder, f'${group_id}.${module}')
+                    # regenerate
+                    check_call(tsp_cmd, sdk_root)
                 succeeded = True
     except subprocess.CalledProcessError as error:
         error_message = (f'[GENERATE][Error] Code generation failed. tsp-client init fails: {error}\n'
                          'If TypeSpec Validation passes, you can open an issue to https://github.com/Azure/autorest.java/issues. Please include the link of this Pull Request in the issue.')
         logging.error(error_message)
         print(error_message, file=sys.stderr)
 
-    if succeeded:
-        # check require_sdk_integration
-        cmd = ['git', 'add', '.']
-        check_call(cmd, sdk_root)
-        cmd = ['git', 'status', '--porcelain', os.path.join(sdk_folder, 'pom.xml')]
-        logging.info('Command line: ' + ' '.join(cmd))
-        output = subprocess.check_output(cmd, cwd=sdk_root)
-        output_str = str(output, 'utf-8')
-        git_items = output_str.splitlines()
-        if len(git_items) > 0:
-            git_pom_item = git_items[0]
-            # new pom.xml implies new SDK
-            require_sdk_integration = git_pom_item.startswith('A ')
-
-        # parse service and module
-        match = re.match(r'sdk[\\/](.*)[\\/](.*)', sdk_folder)
-        service = match.group(1)
-        module = match.group(2)
-
     return succeeded, require_sdk_integration, sdk_folder, service, module
 
 
 def check_call(cmd: List[str], work_dir: str):
     logging.info('Command line: ' + ' '.join(cmd))
     subprocess.check_call(cmd, cwd=work_dir)
 
+def drop_changes(work_dir: str):
```

Comment:
check lint, I assume 2 newlines?

Reply:
[PEP8](https://peps.python.org/pep-0008/#blank-lines) does say that for top level functions, it should be surrounded by two blank lines. Though strangely pylint doesn't complain about this..

Reply:
updated to two blank lines